### PR TITLE
Fix #273: Add 'auto_print' option to automatically print a picture

### LINKED ISSUE #273: Automatically print a photo, and not display the prompt
Printing question: is there a way to automatically print a photo, and not display the prompt?

I'd prefer not to wire up an additional button or UI for folks, and just have a single print be automatic; similar to the way "traditional" photo booths work.

_Originally posted by @mattdsteele in https://github.com/pibooth/pibooth/issues/42#issuecomment-875744758_

### PR DESCRIPTION
**Pull request by anxuae**
@mattdsteele please can you test and give feedback?

### CODE REVIEW

File: docs/sources/config/default.cfg [hunk 1]
```diff
@@ -123,6 +123,9 @@ printer_options = {}
 # How long is the print view in seconds (0 to skip it)
 printer_delay = 10
 
+# How many pages are automatically sent to the printer
```
Comment (werdeil) [inline docs/sources/config/default.cfg:126]:
```suggestion
# Number of pages are automatically sent to the printer
```

File: pibooth/config/parser.py [hunk 2]
```diff
@@ -200,6 +200,10 @@ def values_list_repr(values):
                 (10,
                  "How long is the print view in seconds (0 to skip it)",
                  "Time to show print screen", [str(i) for i in range(0, 21)])),
+            ("auto_print",
+                (0,
+                 "How many pages are automatically sent to the printer",
```
Comment (werdeil) [inline pibooth/config/parser.py:205]:
```suggestion
                 "Number of pages are automatically sent to the printer",
```

File: pibooth/plugins/printer_plugin.py [hunk 3]
```diff
@@ -36,24 +43,19 @@ def state_wait_do(self, cfg, app, events):
...
 
     @pibooth.hookimpl
     def state_processing_enter(self, cfg, app):
         app.count.remaining_duplicates = cfg.getint('PRINTER', 'max_duplicates')
 
+    @pibooth.hookimpl
+    def state_processing_do(self, cfg, app):
+        if app.previous_picture_file and app.printer.is_available() and app.count.remaining_duplicates > 0:
+            for i in range(cfg.getint('PRINTER', 'auto_print')):
+                self.print(cfg, app)
```
Comment (werdeil) [inline pibooth/plugins/printer_plugin.py:56]: I wouldn't have put the auto print in the processing state, why don't you put is in the `state_print_enter`?

**Review by werdeil — CHANGES_REQUESTED** [review]
One comment and one proposal of wording change.

**Comment by mattdsteele** [comment]
Cursory testing of this worked out great, I was able to set `auto_print = 1` and it did just that! 👍

File: pibooth/plugins/printer_plugin.py (hunk 3 above)
Comment (anxuae) [reply pibooth/plugins/printer_plugin.py:56]: The `print` screen can be skipped if delay is = 0 ,  file shall be sent to printer before `print` state

Comment (werdeil) [reply pibooth/plugins/printer_plugin.py:56]: Indeed, maybe in `state_processing_exit` hook then? We already have the count increase using this hook for the same reason (print state can be skipped)

**Review by werdeil — COMMENTED** [review]
I let you choose between `state_processing_do` (and the if `app.previous_picture` exists or the `state_processing_exit`

**Comment by anxuae** [comment]
Not yet tested, I will give you feedback when tested

**Comment by anxuae** [comment]
Can not put it in `state_processing_exit` because it is executed after the `state_processing_validate` where the number of printed pages is checked.

Thus the only place where picture can be automatically printed is in the `state_processing_do`

**Review by werdeil — APPROVED** [review]
Approved then, one thing though that came to my mind: maybe change the name of the function `print` to `print_picture` to avoid any confusion with python's `print` function

**Pull request merged**